aips/aip-x.md

```diff
@@ -0,0 +1,237 @@
+title: Permissioned Signer
+author: runtian-zhou
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Framework
+created: 09/26/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Permissioned Signer
+
+## Summary
+
+Improve the security and usability of signer by allowing to create a permissioned mode signer where users can specify the permissions associated with those signers. Operations that violates those permission setups would be rejected in the transaction.
+
+### Out of scope
+
+We are trying to define a permission standard to specify permissions in the current aptos framework. Despite such standard could be extended to any modules on blockchain, we will be limiting our focus on the aptos framework only so that we can iterate on this standard faster.
+
+## High-level Overview
+
+Right now Move on Aptos has one single permission that is represented as signer. 
+
+- Smart contract uses `signer` to identify who authenticate this operation. Typical code would look like `let addr = signer::address_of(x)`  and use `addr`  as the issuer for this operation.
+    - e.g: framework code  will use `signer`  to determine which account issued the withdraw
+- Smart contract needs `signer` to move items into/away from account.
+
+This results in a huge security concern:
+
+- Different smart contract uses the same `signer`  to authenticate an operation. This means that a malicious smart contract can pass the `signer`  to another module and perform the operation on user’s behalf without being acknowledged by the user.
+- `signer`  could also be abused to inflate your account.
+
+We need to design a solution such that:
+
+- Move modules should be able to
+    - define permissions
+    - Framework/transaction can set those permissions
+- Transactions that tries to execute operations that are not included in the list needs to be aborted and rejected by the Aptos VM.
+    - Work with our existing signer module so that users can still set constraints on what transactions invoking existing on chain move functions could invoke.
+
+
+## Impact
+
+This would be a full compatible change to the framework. Existing on chain move modules wouldn't be impacted. We will add those constraints to the Aptos Framework around popular asset types, i.e: Coin/FungibleAsset/Object/NFT.
+
+## Alternative Solutions
+
+Another way to enforce permission is to leverage AIP-56 the runtime access control engine in Move. However, the runtime access control is more suitable to provide coarse grained permissions around modules but it's hard to represent permissions such as withdrawing X APT from a signer. However, we are open to encoperate the changes to the Move on Aptos language extensions in the future if we want to extend the permission system to the general public.
```

Comment:
Last sentence seems unrelated to the first two, maybe should be a separate paragraph?